spec/surfnperf_spec.js

```diff
@@ -1096,5 +1096,136 @@ define('spec/surfnperf_spec', [
+        }
+      });
+    });
+
+    describe('#getFirstPaint', function() {
+      describe('when the client is using Chrome', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn({
+            firstPaintTime: 1446480965.123456
+          });
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(SurfNPerf.duration('navigationStart', 'chromePaint'));
+        });
+      });
+
+      describe('when the client is using IE/Edge', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(1776480965.123457);
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(SurfNPerf.duration('navigationStart', 'msPaint'));
+        });
+      });
+
+      describe('when the client is using browser without Time to First Paint built in support', function() {
+        it('returns null since the browser does not have built in support', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+ 
+          expect(SurfNPerf.getFirstPaint()).toEqual(null);
+        });
+      });
+    });
+
+    describe("#getFirstPaintFrame", function() {
+      describe('when the client does not support window.requestAnimationFrame', function() {
+        it('returns null', function() {
+          expect(SurfNPerf.getFirstPaintFrame()).toEqual(null);
+        });
+      });
+
+      describe('when the client supports window.requestAnimationFrame but it has not yet been called', function() {
+        it('returns undefined', function() {
+          if(!window.requestAnimationFrame) {
+            window.requestAnimationFrame = function() {};
+          }
+          expect(SurfNPerf.getFirstPaintFrame()).toEqual(undefined);
+        });
+      });
+
+      describe('when the client supports window.requestAnimationFrame and it has been called', function() {
+        beforeEach(function() {
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 1388595600000
+          });
+          SURF_N_PERF = {
+            marks: {
+              pageStart: 1388595604025,
+              firstPaintFrame: 1777595605005
+            },
+            highResMarks: {
+              firstPaintFrame: 3.6180339887
+            }
+          };
```

Comment:
add these spies as well to help make sure each scenario below is using the correct values for its calculations:
```js
          spyOn(SurfNPerf, '_setMeasure');
          spyOn(SurfNPerf, '_getMeasureDuration').andReturn(1234.12345);
```